tests/components/zeroconf/test_init.py

```diff
@@ -175,6 +173,20 @@ async def test_setup_with_ipv6_default(hass, mock_zeroconf):
     assert mock_zeroconf.called_with()
 
 
+async def test_service_with_invalid_name(hass, mock_zeroconf, caplog):
+    """Test we do not crash on service with an invalid name."""
+    with patch.object(
+        zeroconf, "HaServiceBrowser", side_effect=service_update_mock
+    ) as mock_service_browser:
+        mock_zeroconf.get_service_info.side_effect = BadTypeInNameException
+        assert await async_setup_component(hass, zeroconf.DOMAIN, {zeroconf.DOMAIN: {}})
+        hass.bus.async_fire(EVENT_HOMEASSISTANT_STARTED)
+        await hass.async_block_till_done()
+
+    assert len(mock_service_browser.mock_calls) == 1
+    assert "Failed to get info for device name" in caplog.text
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)